data_structures/src/superblock.rs

```diff
@@ -257,11 +264,41 @@ impl SuperBlockState {
         self.received_superblocks.contains(sbv)
     }
 
-    /// Updates the current superblock signing committee for a given superblock index
-    /// #FIXME This function should be update with issue 1395
-    pub fn update_superblock_signing_committee(&mut self, _current_index: u32) {
-        self.current_signing_committee = self.previous_ars_identities.clone();
+    /// Updates the current superblock signing committee for a given superblock hash
+    pub fn update_superblock_signing_committee(
+        &mut self,
+        _signing_committee_size: u32,
+    ) -> Option<HashSet<PublicKeyHash>> {
+        // If the number of identities is lower than committee_size all the members of the ARS sign the superblock
+        let ars_ordered = &self.previous_ordered_ars_identities;
+        if ars_ordered.len() < usize::try_from(_signing_committee_size).unwrap() {
+            self.current_signing_committee = self.previous_ars_identities.clone();
+            self.current_signing_committee.clone()
+        } else {
+            // Get the number of subsets of _signing_committee_size members
+            let n = ars_ordered.len() as u64 / u64::from(_signing_committee_size);
+            // Start counting the members of the subset from the superblock_hash
+            let a = u64::from(*self.current_superblock_hash.as_ref().get(0).unwrap());
+            let b = a % u64::from(_signing_committee_size);
+            let first_member = b % n;
```

Comment:
I have the concern that applying modulo on a uniform number makes the resulting number non-uniform. For example, `x % 100 for x in [0, 255]` gives 2 points for `[0, 199]` but 3 for `[200, 255]`:

![image](https://user-images.githubusercontent.com/1316439/88161549-68527980-cc10-11ea-8409-c7ca130c2ccd.png)

This may translate into some positions in the ARS being "luckier" than others.

Reply:
Here's a good example about modulo bias:
http://martin.upspin.org/2017/12/14/modulo-bias.html

Reply:
Properly speaking, bias is introduced whenever you want to map from set A to set B where the cardinality of set B is not a factor or multiple of set A :sunglasses:

Reply:
I believe this snippet actually gives uniform values in a specific range out of uniform values in a different range:
```rust
use rand::{Rng, rngs::StdRng};
use rand_core::SeedableRng;

pub fn main() {
    let hash = [0;32];
    let rng = &mut StdRng::from_seed(hash);
    let offset = rng.gen_range(0, 255);
    assert_eq!(offset, 173);
}
```

Reply:
But the problem is that that code is consensus-critical but hard to port to other platforms (very Rust specific), so we may have to find something else.

Reply:
The magic_partition function has been changed and it shouldn't have that problem anymore. As an example, now given a vector [0, 1, 2, 3, 4], if you specify to have a subset starting from 3 and length 2, the function returns the subset [3, 0].